opus/application/static_media/js/widgets.js

```diff
@@ -696,13 +866,215 @@ var o_widgets = {
+                $(`#widget__${slug} .op-search-inputs-set:not(:last)`).append(orLabel);
+
+                if ($(`#widget__${slug} .op-add-inputs`).length > 0) {
+                    addInputIcon = $(`#widget__${slug} .op-add-inputs`).detach();
+                }
+
+                o_widgets.attachAddInputIcon(slug, addInputIcon);
+
+            }
+
             opus.widgetsDrawn.unshift(slug);
             o_widgets.customWidgetBehaviors(slug);
             o_widgets.scrollToWidget(widget);
             o_search.getHinting(slug);
         }); // end callback for .done()
     }, // end getWidget function
 
+    renumberInputsAttributes: function(slug) {
+        /**
+         * Reorder all inputs attributes in numerical order when there
+         * are multiple sets of search inputs in one single widget.
+         */
+        let widgetInputs = $(`#widget__${slug} input`);
+        if (widgetInputs.hasClass("RANGE")) {
+            let extraSearchInputs = $(`#widget__${slug} .op-extra-search-inputs`);
+            let minRangeInputs = $(`#widget__${slug} input.op-range-input-min`);
+            let maxRangeInputs = $(`#widget__${slug} input.op-range-input-max`);
+            let qtypes = $(`#widget__${slug} select`);
+
+            let trailingCounter = 0;
+            let trailingCounterString = "";
+            let minInputNamesArray = [];
+
+            let originalMinName = `${$(minRangeInputs).data("slugname")}1`;
+            let originalMaxName = `${$(maxRangeInputs).data("slugname")}2`;
+            let preprogrammedRangesInfo = $(`#widget__${slug} .op-preprogrammed-ranges`);
+            // If there are extra sets of RANGE inputs, we reorder the following:
+            // 1. name attribute for min & max inputs.
+            // 2. attributes & id for customized ranges dropdown lists (this is required for them
+            // to work properly).
+            // 3. attributes for qtypes
+            if (extraSearchInputs.length > 0) {
+                // Renumber min inputs.
+                // The data will be used to set as data-mininput for corresponding ranges collapsible
+                // (categories) containers. That way each specific min input will be linked to its own
+                // ranges info dropdown.
+                minInputNamesArray = o_widgets.renumberAttributesOfSearchElements(minRangeInputs, true);
+
+                // Renumber max inputs.
+                o_widgets.renumberAttributesOfSearchElements(maxRangeInputs);
+
+                // Renumber preprogrammed ranges dropdown.
+                trailingCounter = 0;
+                if (preprogrammedRangesInfo.length > 1) {
+                    for (const eachRangeDropdown of preprogrammedRangesInfo) {
+                        let rangesDropdownCategories = $(eachRangeDropdown).find("li");
+                        trailingCounter++;
+                        trailingCounterString = (`${trailingCounter}`.length === 1 ?
```

Comment:
See earlier comment on how to zero-pad a number.
Actually given that this happens in multiple places it should really be a function in utils.